multimedia/VLC/Portfile

```diff
@@ -40,434 +30,673 @@ subport             lib${name} {
+                ${worksrcpath}/extras/package/macosx/package.mak
+
+            reinplace "s:LD_LIBRARY_PATH:DYLD_LIBRARY_PATH:g" \
+                ${worksrcpath}/Makefile.am
+
+            reinplace "/argv/s/environ/*_NSGetEnviron()/" \
+                ${worksrcpath}/modules/misc/inhibit/xdg.c \
+                ${worksrcpath}/modules/stream_filter/decomp.c
+
+            reinplace "s/extern char \\*\\*environ;/#include <crt_externs.h>/" \
+                ${worksrcpath}/modules/misc/inhibit/xdg.c \
+                ${worksrcpath}/modules/stream_filter/decomp.c
+
+            if {![file exists ${worksrcpath}/contrib/BGHUDAppKit.framework]} {
+                ln -s ${frameworks_dir}/BGHUDAppKit.framework ${worksrcpath}/contrib/BGHUDAppKit.framework
+            }
+
+            # To trick configure
+            file mkdir "${worksrcpath}/contrib/lib"
+        }
+
+        post-destroot {
+            eval file delete [glob ${destroot}${prefix}/lib/vlc/plugins/*/*.la]
+
+            if {[variant_isset qt4] || [variant_isset qt5] || [variant_isset quartz]} {
+                if {${subport} ne "lib${name}"} {
+                    copy ${worksrcpath}/VLC.app ${destroot}${applications_dir}/VLC.app
+
+                    # These are already in ${prefix}, so we don't need to bundle them as well
+                    delete ${destroot}${applications_dir}/VLC.app/Contents/Frameworks
+                    delete ${destroot}${applications_dir}/VLC.app/Contents/lib
+
+                    # There's no need to install these into the bundle and the prefix
+                    delete ${destroot}${applications_dir}/VLC.app/Contents/MacOS/include
+                    delete ${destroot}${applications_dir}/VLC.app/Contents/MacOS/lib
+                    delete ${destroot}${applications_dir}/VLC.app/Contents/MacOS/plugins
+                    delete ${destroot}${applications_dir}/VLC.app/Contents/MacOS/share/locale
+                    delete ${destroot}${applications_dir}/VLC.app/Contents/MacOS/share/lua
+
+                    # http://trac.macports.org/ticket/35131
+                    ln -s ${prefix}/lib ${destroot}${applications_dir}/VLC.app/Contents/MacOS/lib
+                    ln -s ${prefix}/lib/vlc/plugins ${destroot}${applications_dir}/VLC.app/Contents/MacOS/plugins
+                    ln -s ${prefix}/lib/vlc/lua ${destroot}${applications_dir}/VLC.app/Contents/MacOS/share/lua
+                    ln -s ${prefix}/share/locale ${destroot}${applications_dir}/VLC.app/Contents/MacOS/share/locale
+
+                    # the vlc executable needs to be started with a full path to the app bundle executable
+                    # or else the Mac OS X interface will hang beyond even a ^C or ^\ :
+                    move ${destroot}${prefix}/bin/vlc ${destroot}${prefix}/bin/vlc.exe
+                    system "echo \"#!/bin/sh\nexec \\\"${applications_dir}/VLC.app/Contents/MacOS/VLC\\\" \\\"\\\$\@\\\"\" > ${destroot}${prefix}/bin/vlc"
```

Comment:
Yes. I adopted the system echo trick from one of Mojca's contributions to my audacity port.